### PR TITLE
Update software dependencies and fix broken JSON schema

### PR DESCRIPTION
**Pull request by drpatelh**
- [x] Updated software dependencies to latest for all tools:
  - `salmon=1.6.0` -> `salmon=1.10.2`
  - `fastqc=0.11.9` -> `fastqc=0.12.1`
  - `multiqc=1.11` -> `multiqc=1.15`
- [x] Use `conda.channels = 'conda-forge,bioconda,defaults'` in favour of channel pinning in modules
- [x] Added default `profiles` for:
  - `singularity`
  - `apptainer`
  - `mamba`
- [x] Updated broken `nextflow_schema.json`

- [ ] Need to build and push Docker container with new dependencies @pditommaso 
- [ ] Update container version in pipeline

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -38,6 +38,28 @@ profiles {
     process.container = 'quay.io/nextflow/rnaseq-nf:v1.1'
     docker.enabled = true
   }
+
+  singularity {
+    process.container = 'quay.io/nextflow/rnaseq-nf:v1.1'
```
Comment (ewels) [inline nextflow.config:43]: Can we not just define this once, at the top scope?

File: nextflow.config [hunk 2]
```diff
@@ -38,6 +38,28 @@ profiles {
...
+  }
+
+  apptainer {
+    process.container = 'quay.io/nextflow/rnaseq-nf:v1.1'
+    apptainer.enabled = true
+  }
+
+  conda {
+    conda.enabled = true
+    conda.channels = 'conda-forge,bioconda,defaults'
```
Comment (ewels) [inline nextflow.config:55]: As with container, could just be in the top scope once?

File: conda.yml [hunk 3]
```diff
@@ -1,10 +1,9 @@
 name: rnaseq-nf
```
Comment (ewels) [inline conda.yml:1]: Is this file needed any more? Each process now has a single conda package specified, and the `Dockerfile-mamba` hardcodes all the package names..

**Review by ewels — APPROVED** [review]
A few minor suggestions, but generally LGTM to merge even without those 👍🏻

File: nextflow.config (hunk 1 above)
Comment (drpatelh) [reply nextflow.config:43]: I did think about that too but I assumed they were added for a reason. For example, isolating the `container` directive when `conda` is being used.

File: nextflow.config (hunk 2 above)
Comment (drpatelh) [reply nextflow.config:55]: Same with this - might be better practice to set these in profiles to isolate the `conda` settings when `docker` / `singularity` .... are being used.

File: nextflow.config (hunk 1 above)
Comment (ewels) [reply nextflow.config:43]: I think it's just ignored when conda is being used. [This is the DSL1 setup](https://github.com/nf-core/methylseq/blob/03972a686bedeb2920803cd575f4d671e9135af0/nextflow.config#L87-L89) that we used to have in nf-core and it worked fine.

File: conda.yml (hunk 3 above)
Comment (drpatelh) [reply conda.yml:1]: Yep it is being used here still:

https://github.com/nextflow-io/rnaseq-nf/blob/88b8ef803a02137750b8c5e063388e8d6a58ac36/docker/Dockerfile#L3

https://github.com/nextflow-io/rnaseq-nf/blob/88b8ef803a02137750b8c5e063388e8d6a58ac36/docker/Makefile#L6

File: nextflow.config (hunk 1 above)
Comment (drpatelh) [reply nextflow.config:43]: You are right, it is ignored but it may be better practice not to have to ignore it and set in in the correct locations?

File: conda.yml (hunk 3 above)
Comment (ewels) [reply conda.yml:1]: Ah ok, might be worth making `Dockerfile-mamba` do the same thing for consistency then..

**Comment by pditommaso** [comment]
Thanks mate! merged manually [7cecb02](https://github.com/nextflow-io/rnaseq-nf/commit/7cecb026e2b2d2909a8afb68d1d8dd35a448914f)
